be/src/exprs/agg/distinct.h

```diff
@@ -103,86 +104,174 @@ struct DistinctAggregateState<LT, SumLT, FixedLengthLTGuard<LT>> {
+                assert(pos != nullptr);
+                memcpy(pos, key.data, key.size);
+                ctor(pos, key.size, key.hash);
+                distinct_size++;
+                try_convert_to_two_level(mem_pool);
+            });
+        } else {
+#if defined(__clang__) && (__clang_major__ >= 16)
+            two_level_set->lazy_emplace(key, [&](const auto& ctor) {
+#else
+            two_level_set->template lazy_emplace(key, [&](const auto& ctor) {
+#endif
+                uint8_t* pos = mem_pool->allocate_with_reserve(key.size, SLICE_MEMEQUAL_OVERFLOW_PADDING);
+                assert(pos != nullptr);
+                memcpy(pos, key.data, key.size);
+                ctor(pos, key.size, key.hash);
+                distinct_size++;
+            });
+        }
     }
 
-    void update_with_hash(MemPool* mem_pool, Slice raw_key, size_t hash) {
+    void lazy_emplace_with_hash(MemPool* mem_pool, Slice raw_key, size_t hash) {
         KeyType key(reinterpret_cast<uint8_t*>(raw_key.data), raw_key.size, hash);
+        if (set != nullptr) {
+#if defined(__clang__) && (__clang_major__ >= 16)
+            set->lazy_emplace_with_hash(key, hash, [&](const auto& ctor) {
+#else
+            set->template lazy_emplace_with_hash(key, hash, [&](const auto& ctor) {
+#endif
+                uint8_t* pos = mem_pool->allocate_with_reserve(key.size, SLICE_MEMEQUAL_OVERFLOW_PADDING);
+                assert(pos != nullptr);
+                memcpy(pos, key.data, key.size);
+                ctor(pos, key.size, key.hash);
+                distinct_size++;
+                try_convert_to_two_level(mem_pool);
+            });
+        } else {
 #if defined(__clang__) && (__clang_major__ >= 16)
-        set.lazy_emplace_with_hash(key, hash, [&](const auto& ctor) {
+            two_level_set->lazy_emplace_with_hash(key, hash, [&](const auto& ctor) {
 #else
-        set.template lazy_emplace_with_hash(key, hash, [&](const auto& ctor) {
+            two_level_set->template lazy_emplace_with_hash(key, hash, [&](const auto& ctor) {
 #endif
-            uint8_t* pos = mem_pool->allocate_with_reserve(key.size, SLICE_MEMEQUAL_OVERFLOW_PADDING);
-            assert(pos != nullptr);
-            memcpy(pos, key.data, key.size);
-            ctor(pos, key.size, key.hash);
-        });
+                uint8_t* pos = mem_pool->allocate_with_reserve(key.size, SLICE_MEMEQUAL_OVERFLOW_PADDING);
+                assert(pos != nullptr);
+                memcpy(pos, key.data, key.size);
+                ctor(pos, key.size, key.hash);
+            });
+        }
     }
 
-    int64_t disctint_count() const { return set.size(); }
+    void prefetch_hash(size_t hash_value) {
+        if (set != nullptr) {
+            set->prefetch_hash(hash_value);
+        } else {
+            two_level_set->prefetch_hash(hash_value);
+        }
+    }
 
-    size_t serialize_size() const {
+    int64_t serialize_size() const {
```

Comment:
`size_t` to prevent any conversion in future on implicit conversion 

```
int64_t serialize_size() const {
   size_t size = 0;
...
   return size;
}
```

and later
```
    size_t serialize_size() const { return set.serialize_size(); }
```

Reply:
Good suggestion, i will optimize it later.